sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client_connection_async.py

```diff
@@ -3034,13 +3034,14 @@ async def _AddPartitionKey(self, collection_link, document, options):
         # TODO: Refresh the cache if partition is extracted automatically and we get a 400.1001
 
         # If the document collection link is present in the cache, then use the cached partitionkey definition
-        if collection_link in self.partition_key_definition_cache:
-            partitionKeyDefinition = self.partition_key_definition_cache.get(collection_link)
+        if collection_link in self.container_properties_cache:
+            cached_collection = self.container_properties_cache.get(collection_link)
+            partitionKeyDefinition = cached_collection.get("partitionKey")
         # Else read the collection from backend and add it to the cache
         else:
             collection = await self.ReadContainer(collection_link)
             partitionKeyDefinition = collection.get("partitionKey")
-            self.partition_key_definition_cache[collection_link] = partitionKeyDefinition
+            self.container_properties_cache[collection_link] = collection
```

Comment:
same question as above, should this be _set_properties_cache(collection)? (since we only persist few properties in the cache)

Reply:
Yes, thank you for catching that